internal/store/utils.go

```diff
@@ -78,15 +78,22 @@ func kubeLabelsToPrometheusLabels(labels map[string]string) ([]string, []string)
 
 func mapToPrometheusLabels(labels map[string]string, prefix string) ([]string, []string) {
 	labelKeys := make([]string, 0, len(labels))
-	for k := range labels {
-		labelKeys = append(labelKeys, k)
-	}
-	sort.Strings(labelKeys)
-
 	labelValues := make([]string, 0, len(labels))
-	for i, k := range labelKeys {
-		labelKeys[i] = prefix + "_" + sanitizeLabelName(k)
-		labelValues = append(labelValues, labels[k])
+
+	sortedKeys := make([]string, 0)
+	for key := range labels {
+		sortedKeys = append(sortedKeys, key)
+	}
+	sort.Strings(sortedKeys)
+
+	seen := make(map[string]struct{})
+	for _, k := range sortedKeys {
+		labelKey := prefix + "_" + sanitizeLabelName(k)
+		if _, ok := seen[labelKey]; !ok {
+			labelKeys = append(labelKeys, labelKey)
+			labelValues = append(labelValues, labels[k])
+			seen[labelKey] = struct{}{}
+		}
```

Comment:
I would like to add a log line here for debugging purposes that prints which ones are thrown away. WDYT?

Reply:
I was worried that it might get too spammy as this code triggers multiple times per second.
So I'll let you folks decide, just let me how should I approach it.

Reply:
I was thinking of log level debug, so folks can enable it if they need to debug something.

Reply:
@lilic, I've added basic logging, but it might have some issues:

1. This is the first time klog is being used inside `internal`, so I wasn't sure if that's okay. I also wasn't sure about the exact verbosity level, so I've set it at 4 for now.
2. The context is limited to what's available inside `mapToPrometheusLabels`, so currently I can only log the label itself before and after sanitization. I could also log the value, but I don't think that's useful. We could add more context by passing extra data about resource type and resource name inside `mapToPrometheusLabels ` / `kubeLabelsToPrometheusLabels` to make logs more useful.

Please let me know if I should change anything.